edward/stats/distributions.py

```diff
@@ -9,114 +9,83 @@ class Distribution:
-        SciPy's naming and argument conventions.
-
-        The equivalent method in SciPy is not guaranteed to be
-        supported with a batch of parameter inputs, e.g., a vector of
-        location parameters in a normal distribution, or a matrix of
-        concentration parameters in a Dirichlet. This is.
-
-        This does not follow SciPy's behavior, e.g., the number (or
-        shape) of the draws will always be denoted by its outer
-        dimension(s).
-
-        params as a 2d or higher tensor is not guaranteed to be
-        supported (for either univariate or multivariate
-        distribution).
-
-        size as a list or tuple of more than one element is not
-        guaranteed to be supported.
-
-        For most distributions, the parameters must be of the same
-        shape and type, e.g., n and p in Binomial must be np.arrays()
-        of same shape or both floats. For some, they may differ by one
-        dimension, e.g., n and p in Multinomial can be float and
-        np.array(), or both np.arrays, and n always has one less
-        dimension.
+        many distributions for random number generation.
         """
         raise NotImplementedError()
 
     def logpmf(self, x):
         """
         Parameters
         ---------
-        x : float, np.array, or tf.Tensor
-            If univariate distribution, can be a scalar or tensor.
-            If multivariate distribution, can be a tensor; the outer
-            dimension carries the multivariate dimension.
+        x : np.array or tf.Tensor
+            If univariate distribution, can be a scalar, vector, or matrix.
+            If multivariate distribution, can be a vector or matrix.
 
-        params : float, np.array, or tf.Tensor
+        params : np.array or tf.Tensor
             scalar unless documented otherwise
 
         Returns
         -------
         tf.Tensor
-            If univariate distribution, returns a tensor of same shape
-            as input. If multivariate distribution, returns a tensor
-            of shape[:-1] from input: the outer dimension representing
-            the multivariate dimension is collapsed.
-
-        Notes
-        -----
-        x as a 3d or higher tensor is not guaranteed to be supported
-        (for either univariate or multivariate distribution).
+            For univariate distributions, returns a scalar, vector, or
+            matrix corresponding to the size of input. For
+            multivariate distributions, returns a scalar if vector
+            input and vector if matrix input, where each element in
+            the vector evaluates a row in the matrix.
         """
         raise NotImplementedError()
 
     def entropy(self):
         """
         Parameters
         ---------
-        params : float, np.array, or tf.Tensor
-            If univariate distribution, can be a scalar, vector, or
-            matrix.
+        params : np.array or tf.Tensor
```

Comment:
This should keep `float, np.array, or tf.Tensor`. parameters of type `float` is supported